prometheus-to-sd/main.go

```diff
@@ -42,18 +43,36 @@ var (
 		"Comma-separated list of whitelisted metrics. If empty all metrics will be exported.")
 	apioverride = flag.String("api-override", "",
 		"The stackdriver API endpoint to override the default one used (which is prod).")
+	source = flags.Uris{}
 )
 
 func main() {
 	flag.Set("logtostderr", "true")
+	flag.Var(&source, "source", "source(s) to watch in [component]:http://host:port?whitelisted=a,b,c format")
+
 	defer glog.Flush()
 	flag.Parse()
 
-	if *component == "" {
-		glog.Fatalf("--component flag is required")
+	var sourceConfigs []config.SourceConfig
+
+	for _, c := range source {
+		if sourceConfig, err := config.ParseSourceConfig(c); err != nil {
+			glog.Fatalf("Error while parsing source config flag %v: %v", c, err)
+		} else {
+			sourceConfigs = append(sourceConfigs, *sourceConfig)
+		}
 	}
 
-	glog.Infof("Running prometheus-to-sd, monitored target is %s %v:%v", *component, *host, *port)
+	if *component != "" {
```

Comment:
I'd allow component (and port) only where there is no sources specified.
Also let's print a warning somewhere that it's deprecated.